client/release.go

```diff
@@ -155,26 +155,15 @@ func (c *Client) PromoteRelease(appID string, appType string, sequence int64, la
 	return errors.New("unknown app type")
 }
 
-// yamlOrJSON is either a single ship yaml file, or a serialized JSON object describing a yaml-dir, created by readYAMLDir()
+// data is a []byte describing a tarred yaml-dir, created by tarYAMLDir()
 // this Client abstraction continue to spring more leaks :)
-func (c *Client) LintRelease(appID string, appType string, yamlOrJSON string) ([]types.LintMessage, error) {
+func (c *Client) LintRelease(appType string, data []byte) ([]types.LintMessage, error) {
 
 	if appType == "platform" {
 		return nil, errors.New("Linting is not yet supported in this CLI, please install github.com/replicatedhq/replicated-lint to lint this application")
-		// return c.PlatformClient.LintRelease(appID, yamlOrJSON)
-	} else if appType == "ship" {
-		return c.ShipClient.LintRelease(appID, yamlOrJSON)
```

Comment:
Rather than removing this branch, can we return an error about how Ship linting is not supported?

Reply:
yep, for sure